### PR TITLE
Added unit tests for JDBC module

### PR DESCRIPTION
**Pull request by Trianz-Akshay**
*Issue #, if available:*

*Description of changes:*
Added unit test cases for JDBC module.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by burhan94** [comment]
Can we consider using GenAI here to add even more unit tests? I'd like to see 30-50 UT's get generated that test the different expressions and the resulting translations

**Comment by ritiktrianz** [comment]
Hi @burhan94  ,Thank you for the feedback. We already tried GenAI using Claude Sonnet models, for unit test generation. It generated large volume of test cases covering multiple scenarios, such as edge cases, negative paths, and broader input variations. We felt that not all of these test cases may be relevant and may create unnecessary overhead. So, we removed a few tests and kept only the relevant ones.
 
However, now we have added the generated tests after reviewing them manually.

**Comment by burhan94** [comment]
Can we add some tests for ORDER BY, and then a test that has ORDER BY, LIMIT, and a complex expression? I'd like to also validate that we're testing the different complex expressions supported by the framework.

**Comment by Trianz-Akshay** [comment]
> Can we add some tests for ORDER BY, and then a test that has ORDER BY, LIMIT, and a complex expression? I'd like to also validate that we're testing the different complex expressions supported by the framework.

Thanks for the review! We already had tests for ORDER BY and LIMIT functionality. Now we have added comprehensive tests for complex expressions functionality.

**Review by burhan94 — DISMISSED** [review]

File: athena-jdbc/src/test/java/com/amazonaws/athena/connectors/jdbc/JdbcEnvironmentPropertiesTest.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        String specialSecret = "secret/name#1";
+        connectionProperties.put(SECRET_NAME, specialSecret);
+        
+        Map<String, String> result = jdbcEnvironmentProperties.connectionPropertiesToEnvironment(connectionProperties);
+        String expected = BASE_CONNECTION_STRING + "?${" + specialSecret + "}";
+        assertEquals(expected, result.get(DEFAULT));
+    }
+
+    @Test
+    public void testMultipleSecretsAndParams() {
```
Comment (ejeffrli) [inline athena-jdbc/src/test/java/com/amazonaws/athena/connectors/jdbc/JdbcEnvironmentPropertiesTest.java]: do we allow multiple secrets? I think we do not. Maybe possible to remove this test

File: athena-jdbc/src/test/java/com/amazonaws/athena/connectors/jdbc/manager/JDBCUtilTest.java [hunk 2]
```diff
@@ -121,22 +126,22 @@ public void createJdbcMetadataHandlerMap()
...
+        Assert.assertEquals(FakeDatabaseJdbcMetadataHandler.class, catalogs.get("lambda:functionName").getClass());
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void createJdbcMetadataHandlerMapDifferentDatabasesThrows()
     {
-        Map<String, JdbcMetadataHandler> catalogs = JDBCUtil.createJdbcMetadataHandlerMap(ImmutableMap.<String, String>builder()
+        JDBCUtil.createJdbcMetadataHandlerMap(ImmutableMap.<String, String>builder()
                 .put("testCatalog1_connection_string", CONNECTION_STRING1)
                 .put("testCatalog2_connection_string", CONNECTION_STRING2)
```
Comment (ejeffrli) [inline athena-jdbc/src/test/java/com/amazonaws/athena/connectors/jdbc/manager/JDBCUtilTest.java]: Can we make this test more clear why it throws IllegalArgumentException? I think it is because CONNECTION_STRING1 and CONNECTION_STRING2 are prepended by the wrong engine type.

Comment (ejeffrli) [reply athena-jdbc/src/test/java/com/amazonaws/athena/connectors/jdbc/manager/JDBCUtilTest.java]: Maybe instead of just testCatalog1_connection_string and testCatalog2_connection_String, one can be "wrong_connection_string". 

Please check why this is throwing IllegalArgumentException before

File: athena-jdbc/src/test/java/com/amazonaws/athena/connectors/jdbc/JdbcEnvironmentPropertiesTest.java (hunk 1 above)
Comment (ritiktrianz) [reply athena-jdbc/src/test/java/com/amazonaws/athena/connectors/jdbc/JdbcEnvironmentPropertiesTest.java]: Hi @ejeffrli , thanks for the review comment. We’ve removed this test as suggested.

File: athena-jdbc/src/test/java/com/amazonaws/athena/connectors/jdbc/manager/JDBCUtilTest.java (hunk 2 above)
Comment (ritiktrianz) [reply athena-jdbc/src/test/java/com/amazonaws/athena/connectors/jdbc/manager/JDBCUtilTest.java]: @ejeffrli , We’ve updated the test and verified that exception occurs because one of the provided connection strings is for the "notrealdb" engine, which is not aligned with the "fakedatabase" engine type that the handler is configured for.
